handlers/lookup.go

```diff
@@ -60,6 +60,7 @@ func (l *lookupHandler) handleMissingRoute(rw http.ResponseWriter, r *http.Reque
 	l.reporter.CaptureBadRequest()
 
 	rw.Header().Set("X-Cf-RouterError", "unknown_route")
+	rw.Header().Set("Cache-Control", "no-cache")
```

Comment:
I wonder if a wider set of "do not cache" values in the `Cache-Control` header might be suitable for this default 404 scenario: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cache_response_directives

Reply:
I think using `no-store` instead might be a good bet.

Brief summary:
* `no-store` tells caches not to cache this resource at all.
* `no-cache` tells caches to always check that the E-Tag with upstream, but they're allowed to respond from cache if they get a 304 Not-Modified back.

I can't imagine gorouter is going to be serving 304s for unknown-routes, and in any case the payload is tiny in this case, so probably just disabling caching entirely is the right thing to do.

I think there's no point in adding `must-revalidate` and friends, because they're already covered by `no-cache` / `no-store`.

Reply:
Updated to use `no-store`.